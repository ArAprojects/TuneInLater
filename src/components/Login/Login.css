.account-container {
    background-image: url('book2.jpg');
    background-size: cover;
    min-height: 100vh;
    display:flex;
    justify-content: space-evenly;
    align-items:center;
}

h4 {
    font-size: 1.5em;
}

.create-account-container {
    border-radius: 10px;
    background-color:white;
    opacity: 0.9;
    display: flex;
    width: 500px;
    height: 300px;
    flex-direction: column;
    align-items: center;
    text-align:center;
}

.login-container {
    border-radius: 10px;
    background-color:white;
    opacity: 0.9;
    width: 500px;
    height: 300px;
    display: flex;
    width: 500px;
    flex-direction: column;
    align-items: center;
    text-align:center;
    /* justify-content:space-evenly; */
}

.login-email,
.login-password {
    margin-bottom: 25px
}

input {
    border-radius: 4px;
    display: flex;
    margin-top: 10px;
    width: 350px;
    font-size:1.2em;
    padding-left:6px;
}

.or {
    display: flex;
    align-items: center;
    font-size: 1.5em;
}

button {
    height:30px;
    width: 100px;
    font-size: 1em
}

.create-account-button {
    background-color: #7973b3;
    margin-top: 15px;
    width: 150px;
}

.login-button {
    background-color: #5db9e0;
    width: 150px;
}
