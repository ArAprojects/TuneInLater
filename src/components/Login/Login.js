import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import './Login.css'

class Login extends Component {
  constructor(props) {
    super(props);
    this.state = {
      createName: '',
      createEmail: '',
      createPassword: '',
      loginEmail: '',
      loginPassword: ''
    }
  }

  structureUserObject = (e) => {
    e.preventDefault();
    const { createName, createEmail, createPassword} = this.state;
    const userObject = {
      name: createName,
      email: createEmail,
      password: createPassword
    }
    this.props.createNewUser(userObject)
  }

  structureLoginUserObject = () => {
    const { loginEmail, loginPassword } = this.state;
    const userObject = {
      email: loginEmail,
      password: loginPassword
    }
    this.props.loginUser(userObject)
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value })
  }

  render() {
    console.log(this.props)
    const { createName, createEmail, createPassword, loginEmail, loginPassword } = this.state;
    return (
    <div className="big-container">
      <header>
        <h1>Tune-In-Later</h1>
      </header>
      <section className="account-container">
        <article className="create-account-container">
          <h4>Create An Account</h4>
          <form className="create-account-form">
            <input
              type="text"
              placeholder="Name"
              name="createName"
              value={createName}
              onChange={(e) => this.handleChange(e)}
            />
            <input
              type="text"
              placeholder="Email"
              name="createEmail"
              value={createEmail}
              onChange={(e) => this.handleChange(e)}
            />
            <input
              type="text"
              placeholder="Password"
              name="createPassword"
              value={createPassword}
              onChange={(e) => this.handleChange(e)}
            />
          <button className="create-account-button" onClick={(e) => this.structureUserObject(e)}>Create Account</button>
          </form>
        </article>
        <h4 className="or">OR</h4>
        <article className="login-container">
          <h4>Login</h4>
          <form className="login-form">
            <input
            className="login-email"
              type="text"
              placeholder="Email"
              name="loginEmail"
              value={loginEmail}
              onChange={(e) => this.handleChange(e)}
            />
            <input
            className="login-password"
              type="text"
              placeholder="Password"
              name="loginPassword"
              value={loginPassword}
              onChange={(e) => this.handleChange(e)}
            />
            <NavLink to='/'>
              <button className="login-button" onClick={() => this.structureLoginUserObject()}>Login</button>
            </NavLink>
          </form>
        </article>
      </section>
    </div>
    )
  }
}

export default Login;
