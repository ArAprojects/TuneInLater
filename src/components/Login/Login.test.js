import React from 'react';
import { shallow } from 'enzyme';
import Login from './Login';

describe('Login', () => {
  let wrapper;
  const mockLoginUser = jest.fn();
  const mockCurrentUser = {
    id: 3,
    name: "bill",
    email: "bill@gmail.com"
  }
  const mockCreateNewUser = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Login 
      loginUser={mockLoginUser}
      currentUser={mockCurrentUser}
      createNewUser={mockCreateNewUser}
    />)
  })

  it('should take a snapshot when passed valid data', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should update createName state onChange of input', () => {
    wrapper.find('input').at(0).simulate('change', { target: { value: 'Jacob', name: 'createName' } });
    expect(wrapper.state('createName')).toEqual('Jacob')
  })

  it('should update createEmail state onChange of input', () => {
    wrapper.find('input').at(1).simulate('change', { target: { value: 'jacob@gmail.com', name: 'createEmail' } });
    expect(wrapper.state('createEmail')).toEqual('jacob@gmail.com')
  })

  it('should update createPassword state onChange of input', () => {
    wrapper.find('input').at(2).simulate('change', { target: { value: 'sweetPassword123', name: 'createPassword' } });
    expect(wrapper.state('createPassword')).toEqual('sweetPassword123')
  })

  it('should update loginEmail state onChange of input', () => {
    wrapper.find('input').at(3).simulate('change', { target: { value: 'jogren@gmail.com', name: 'loginEmail' } });
    expect(wrapper.state('loginEmail')).toEqual('jogren@gmail.com')
  })

  it('should update loginPassword state onChange of input', () => {
    wrapper.find('input').at(4).simulate('change', { target: { value: 'sweetPassword321', name: 'loginPassword' } });
    expect(wrapper.state('loginPassword')).toEqual('sweetPassword321')
  })

  it('should invoke createNewUser with a structued object', () => {
    const expected = {
      name: 'Jacob',
      email: 'jacob@gmail.com',
      password: 'sweetPassword123'
    }
    let e = {
      preventDefault: jest.fn()
    }

    wrapper.setState({ 
      createName: 'Jacob',
      createEmail: 'jacob@gmail.com',
      createPassword: 'sweetPassword123'
     })
    wrapper.instance().structureUserObject(e);
    expect(mockCreateNewUser).toHaveBeenCalledWith(expected);
  })

  it('should invoke loginUser with a structured user object', () => {
    const expected = {
      email: 'jacob@gmail.com',
      password: 'sweetPassword123'
    }

    wrapper.setState({
      loginEmail: 'jacob@gmail.com',
      loginPassword: 'sweetPassword123'
    })
    wrapper.instance().structureLoginUserObject();
    expect(mockLoginUser).toHaveBeenCalledWith(expected);
  })

})