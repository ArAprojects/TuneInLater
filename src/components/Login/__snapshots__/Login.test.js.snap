// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should take a snapshot when passed valid data 1`] = `
<div
  className="big-container"
>
  <header>
    <h1
      className="title"
    >
      Tune-In-Later
    </h1>
  </header>
  <section
    className="account-container"
  >
    <article
      className="create-account-container"
    >
      <h4>
        Create An Account
      </h4>
      <form
        className="create-account-form"
      >
        <input
          name="createName"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value=""
        />
        <input
          name="createEmail"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
        <input
          name="createPassword"
          onChange={[Function]}
          placeholder="Password"
          type="text"
          value=""
        />
        <button
          className="create-account-button"
          onClick={[Function]}
        >
          Create Account
        </button>
      </form>
    </article>
    <h4
      className="or"
    />
    <article
      className="login-container"
    >
      <h4>
        Login
      </h4>
      <form
        className="login-form"
      >
        <input
          className="login-email"
          name="loginEmail"
          onChange={[Function]}
          placeholder="Email"
          type="text"
          value=""
        />
        <input
          className="login-password"
          name="loginPassword"
          onChange={[Function]}
          placeholder="Password"
          type="text"
          value=""
        />
        <NavLink
          to="/"
        >
          <button
            className="login-button"
            onClick={[Function]}
          >
            Login
          </button>
        </NavLink>
      </form>
    </article>
  </section>
</div>
`;
